.github/workflows/release.yml

```diff
@@ -0,0 +1,120 @@
+      upload_url: ${{ steps.release.outputs.upload_url }}
+      release_version: ${{ env.RELEASE_VERSION }}
+    steps:
+    - name: Get the release version from the tag
+      shell: bash
+      if: env.RELEASE_VERSION == ''
+      run: |
+        # See: https://github.community/t5/GitHub-Actions/How-to-get-just-the-tag-name/m-p/32167/highlight/true#M1027
+        echo "RELEASE_VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+        echo "version is: ${{ env.RELEASE_VERSION }}"
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 1
+    - name: Create GitHub release
+      id: release
+      uses: actions/create-release@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      with:
+        tag_name: ${{ env.RELEASE_VERSION }}
+        release_name: ${{ env.RELEASE_VERSION }}
+  build-release:
+    name: build-release
+    needs: create-release
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [ubuntu-latest, ubuntu-arm-latest, macOS-latest, macOS-arm-latest, windows-latest]
+        rust: [stable]
+        include:
+          - name: ubuntu-arm-latest
+            os: ubuntu-latest
+            target: linux-aarch64
+            runner: [self-hosted, linux, ARM64]
+          - name: ubuntu-latest
+            os: ubuntu-latest
+            target: linux-x86_64
+            runner: [self-hosted, linux, X64]
+        # TODO: x86_64 runner is not available on the org level
+        #   - name: macOS-latest
+        #     os: macOS-latest
+        #     target: darwin-x86_64
+        #     runner: [self-hosted, macOS, X64]
+          - name: macOS-arm-latest
+            os: macOS-latest
+            target: darwin-aarch64
+            runner: [self-hosted, macOS, ARM64]
+        # TODO: windows runner is not available on the org level
+          - name: windows-latest
+            os: windows-latest
+            target: windows-x86_64
+            runner: [windows-latest]
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 1
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@stable
+      with:
+        toolchain: ${{ matrix.rust }}
+    - name: Build release binary
+      run: cargo build --verbose --release
+    - name: Build archive
+      shell: bash
+      run: |
+        outdir="./target/release"
+        staging="${{ env.BIN_NAME }}-${{ needs.create-release.outputs.release_version }}-${{ matrix.target }}"
+        mkdir -p "$staging"/complete
+        cp {README.md,LICENSE-*} "$staging/"
+        if [ "${{ matrix.os }}" = "windows-latest" ]; then
+          cp "target/release/${{ env.BIN_NAME }}.exe" "$staging/"
+          cd "$staging"
+          7z a "../$staging.zip" .
+          echo "ASSET=$staging.zip" >> $GITHUB_ENV
+        else
+          cp "target/release/${{ env.BIN_NAME }}" "$staging/"
+          tar czf "$staging.tar.gz" -C "$staging" .
```

Comment:
this is probably bikeshedding, but really there's no reason to keep using the old-compatibility arguments to tar and using `-czf` is a better choice these days.  There's generally very low probability that your tar commands will still work on old tar implementations that need those anyway.